python/across_api/auth/api.py

```diff
@@ -56,17 +57,21 @@ async def __call__(self, request: Request) -> None:
                 resp = await client.get(jwks_uri)
             jwks_data = resp.json()
             header = jwt.get_unverified_header(credentials.credentials)
-            for signing_key in jwks_data["keys"]:
-                if signing_key["kid"] == header["kid"]:
-                    break
+
+            signing_key = None
+            for key in jwks_data["keys"]:
+                if "kid" in header.keys():
+                    if key["kid"] == header["kid"]:
+                        signing_key = key
+                        break
             else:
```

Comment:
It's a valid criticism that the `for: else` construct esoteric, but this version is now redundant. Try this instead:

```suggestion
            if signing_key is None:
```